lib/sycamore/sycamore/document.py

```diff
@@ -0,0 +1,388 @@
+
+    def category(self):
+        return Category.Caption
+
+    def accept(self, visitor):
+        return visitor.visit_list_item(self)
+
+
+class PageFooter(Leaf):
+    def __init__(self, node_id, box, content, metadata):
+        super().__init__(node_id, box, content, metadata)
+
+    def category(self):
+        return Category.Caption
+
+    def accept(self, visitor):
+        return visitor.visit_page_footer(self)
+
+
+class PageHeader(Leaf):
+    def __init__(self, node_id, box, content, metadata):
+        super().__init__(node_id, box, content, metadata)
+
+    def category(self):
+        return Category.Caption
+
+    def accept(self, visitor):
+        return visitor.visit_page_header(self)
+
+
+class Picture(Leaf):
+    def __init__(self, node_id, box, content, metadata):
+        super().__init__(node_id, box, content, metadata)
+
+    def category(self):
+        return Category.Caption
+
+    def accept(self, visitor):
+        return visitor.visit_picture(self)
+
+
+class SectionHeader(Leaf):
+    def __init__(self, node_id, box, content, metadata):
+        super().__init__(node_id, box, content, metadata)
+
+    def category(self):
+        return Category.Caption
+
+    def accept(self, visitor):
+        return visitor.visit_section_header(self)
+
+
+class Table(Leaf):
+    def __init__(self, node_id, box, content, metadata, continued=None):
+        super().__init__(node_id, box, content, metadata)
+
+    def category(self):
+        return Category.Caption
+
+    def accept(self, visitor):
+        return visitor.visit_table(self)
+
+
+class Text(Leaf):
+    def __init__(self, node_id, box, content, metadata, continued=None):
+        super().__init__(node_id, box, content, metadata)
+        self._continued = continued
+
+    def category(self):
+        return Category.Caption
+
+    def continued(self):
+        return self._continued
+
+    def accept(self, visitor):
+        return visitor.visit_text(self)
+
+
+class Title(Leaf):
+    def __init__(self, node_id, box, content, metadata):
+        super().__init__(node_id, box, content, metadata)
+
+    def category(self):
+        return Category.Caption
+
+    def accept(self, visitor):
+        return visitor.visit_title(self)
+
+
+class Group(Internal):
+    """
+    Group semantic related objects together, e.g. list items, figure and caption
+    """
+    def __init__(self, node_id, children, metadata):
+        super().__init__(node_id, children, metadata)
+
+    def category(self):
+        return Category.Group
+
+    def accept(self, visitor):
+        return visitor.visit_group(self)
+
+
+class Section(Internal):
+    def __init__(self, node_id, children, header, metadata):
+        super().__init__(node_id, children, metadata)
+        self._header = header
+
+    def category(self):
+        return Category.Section
+
+    def accept(self, visitor):
+        return visitor.visit_section(self)
+
+
+class Visitor(ABC):
```

Comment:
Just curious, what was your rationale for using a visitor like this vs a traverser like we do for plan nodes (https://github.com/aryn-ai/sycamore/blob/9ddcaef57088b4430e02b4ec2aab276d5780a6e5/lib/sycamore/sycamore/plan_nodes.py#L10). To be clear, I don't think I have a strong opinion either way, just curious if you had a reason to prefer one over the other.

Reply:
I'm not aware of this nodetraverse before. I looked a little bit on the printtraverse, seems it still either depend on modifying each node for polymorphism or squeeze in a big function handling all various node types.